src/App/AppZipper.php

```diff
@@ -4,12 +4,40 @@
 
 use RecursiveDirectoryIterator;
 use RecursiveIteratorIterator;
+use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Filesystem\Path;
 use ZipArchive;
 
 class AppZipper
 {
+    private Filesystem $filesystem;
+
+    public function __construct(private string $cacheDir)
+    {
+        $this->filesystem = new Filesystem();
+    }
+
     public function zip(App $app): string
+    {
+        return $this->cache($app);
+    }
+
+    private function cache(App $app): string
+    {
+        $path = Path::join($this->cacheDir, $app->version . '-' . $app->hash . '.zip');
+
+        if (file_exists($path)) {
+            return (string) file_get_contents($path);
+        }
+
+        $result = $this->doZip($app);
+
+        $this->filesystem->dumpFile($path, $result);
```

Comment:
In my opinion, it might be better to pass the $path to the doZip function and remove the file unlink operation. This way, we can also eliminate the redundant filesystem->dumpFile call.

Reply:
i refactored for now to use fs cache